eth/v1alpha1/beacon_chain.proto

```diff
@@ -254,6 +261,64 @@ service BeaconChain {
     } 
 }
 
+message ChainInfo {
```

Comment:
Hi @mcdee my only feedback for these big message types is we use ranges for the protobuf field numbers: (https://developers.google.com/protocol-buffers/docs/proto). The larger the message, the more likely one or more of its fields will change over time (e.g. we may want to include a few more chain configuration values down the line if needed), and we cannot change field numbers in flight. Instead, we can do something like:
```
// Chain configuration parameters, reserved [1-100]
uint64 seconds_per_slot = 1;
uint64 slots_per_epoch = 2;

// Genesis information, reserved [101-200]
bytes genesis_fork_version = 101;
bytes genesis_validators_root = 102;
```
This gives us flexibility in adding/deprecating fields without causing chaos in numbering.

Reply:
I'm not sure I ordered the fields that way in the first place!

Taking a look through the fields, I'm not sure how best to separate them out.  A first pass could be:

  - genesis time and deposit contract address
  - chain configuration parameters
  - fork information

I'm not sure that we'd increase the information available in the first or last category, though, so perhaps if the fields were rearranged so they went genesis time/deposit contract address, fork information, chain configuration parameters and then we could add more chain configuration parameters if/when they are requested?